src/Unirest/Request.php

```diff
@@ -22,6 +21,10 @@ class Request
     private static $overrideRetryForNextRequest = OverrideRetry::USE_GLOBAL_SETTINGS;
     private static $verifyPeer = true;
     private static $verifyHost = true;
+    private static $connectionTimeout = 300;
```

Comment:
I have updated the PR with a few changes, after an in-depth review of the functionality added but I still couldn't understand the use of Timeout header, is it relevant with connections header?